csprng.php

```diff
@@ -81,28 +108,54 @@ function random_hex($stringLength)
  *
  * @return int
  *
- * @throws Exception
+ * @throws Exception, InvalidArgumentException
  */
 function random_int($min, $max)
 {
-    // @todo Validate $min & $max here
-
-    $bytes = random_bytes(256);
-    // @todo Convert bytes to int somehow?
-    // @todo Will seeding mt_rand with pr-bytes from urand make it cryptographically secure?
-    // @todo Does seeding happen globally?
-    mt_srand();
+    $range = $max - $min;
+    if ($range < 1) {
+        throw new InvalidArgumentException('random_int() expects two different integers.');
+    }
 
-    return mt_rand($min, $max);
+    // 7776 -> 13
+    $bits = ceil(log($range)/log(2));
```

Comment:
The patch as been updated with [an algo that's log-free](https://github.com/php/php-src/pull/1119/files#diff-f1067207e863d8fa568e63446920e7fcR109).